tests/Remove-DbaDbMailProfile.Tests.ps1

```diff
@@ -0,0 +1,57 @@
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
+. "$PSScriptRoot\constants.ps1"
+
+Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
+    Context "Validate parameters" {
+        [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
+        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Name', 'ExcludeProfile', 'InputObject', 'EnableException'
+        It "Should only contain our specific parameters" {
+            Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params | Should -BeNullOrEmpty
+        }
+    }
+}
+
+Describe "$CommandName Integration Tests" -Tag "IntegrationTests" {
+
+    BeforeEach {
+
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server2 = Connect-DbaInstance -SqlInstance $script:instance3
```

Comment:
Testing against instance3 is sufficient as there no big difference between 2016 and 2017.

Reply:
The purpose of this test is to check if the command works against multiple instances, as there were commands in the past where supplying multiple instances did not work, i.e. command worked only on the last instance from the list.

Reply:
If you want to test against multiple instances then pass in `instance3` twice 😉

Reply:
How would I know that the command worked twice since it's removing the same object from the same instance? 
Once it's removed there is no way to say, it was removed in the first go or the last go. 

Example:
One of the command (don't remember which one, there is a PR for that in history) worked well only on a single instance. When supplied with instance A and instance B, it only worked on instance B as that was the last one in the loop. 

If I supply Instance A, and Instance A how do I tell if it ran on 
- both
- first only
- second only
With two different instances, I can easily tell that.

Reply:
Removed the second server to save processing time.